modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/data_renderers/ActionsDropdownRenderer.js

```diff
@@ -111,6 +111,71 @@ export function handleAction(
 
 		window.open(url);
 	}
+	else if (target === 'download') {
+		event.preventDefault();
+
+		setLoading(true);
+
+		let filename = '';
+
+		executeAsyncAction(url, method)
+			.then((response) => {
+				const disposition = response.headers.get('Content-Disposition');
+
+				if (disposition && disposition.indexOf('attachment') !== -1) {
+
+					// Captures the filename from the Content-Disposition header.
+					// For example, "attachment; filename="Filename.json" will capture
+					// "Filename.json".
+
+					const filenameRegex = /filename[^;=\n]*=((['"]).*?\2|[^;\n]*)/;
+
+					const matchesRegex = filenameRegex.exec(disposition);
+
+					if (matchesRegex !== null && matchesRegex[1]) {
+						filename = matchesRegex[1].replace(/['"]/g, '');
+					}
+				}
+
+				return response.blob();
+			})
```

Comment:
The purpose of this request seems to be to:
1. get the file name, from header
2. get URL of the actual resource to download

As a consumer of the component API, I would expect that `url` prop is the URL of the resource to download, not URL for this additional request.

Instead of making a request for the file name, can we pass it in as a prop? Something like `action.data.downloadFileName`. Along with it, we can pass in `url` as the actual resource URL.

Reply:
Yea I thought originally the rest endpoint would offer the actual resource to download, as the open API seems to show in the response body, but we ran into those issues with authentication as well as having a Content Type of "application/xml" when trying to open it with `window.open`. Maybe the link to the export URL would help? `SXPElementResourceImpl.java#getSXPElementExport`

![image](https://user-images.githubusercontent.com/14063502/146844717-0a7c2352-84fa-4588-ab9d-3f7d6fb90924.png)

We can maybe discuss the idea of a `downloadFileName` prop with our backend engineer @peerkar, though he is on PTO til January 😅